docs/reference/search/search-fields.asciidoc

```diff
@@ -99,15 +109,82 @@ values copied using the <<copy-to,`copy_to`>> mapping parameter.
 * Since the `_source` is stored as a single field in Lucene, the whole source
 object must be loaded and parsed, even if only a small number of fields are
 needed.
+====
+
+[[docvalue-fields]]
+.*Doc value fields*
+[%collapsible]
+====
+You can use the <<docvalue-fields,`docvalue_fields`>> parameter to return
+<<doc-values,doc values>> for one or more fields in the search response.
+
+Doc values store the same values as the `_source` but in an on-disk,
+column-based structure that's optimized for sorting and aggregations. Because
+they've been normalized, doc values are typically more compact than `_source`
+values.
+
+.*Example*
+[%collapsible]
+=====
+The following search API request uses the `docvalue_fields` parameter to
+retrieve doc values for the following fields:
 
-{es} supports some alternative methods for returning fields that help avoid
-these downsides:
+* Fields with names starting with `my_ip`
+* `my_keyword_field`
+* Fields with names ending with `_date_field`
 
-* The <<request-body-search-docvalue-fields, `docvalue_fields`>>
-parameter allows for loading fields from their doc values. This can be a good
-choice when returning a fairly small number of fields that support doc values,
-such as keywords and dates.
-* It's also possible to store an individual field's values by using the
+[source,console]
+----
+GET /_search
+{
+  "query": {
+    "match_all": {}
+  },
+  "docvalue_fields": [
+    "my_ip*",                     <1>
+    {
+      "field": "my_keyword_field" <2>
+    },
+    {
+      "field": "*_date_field",
+      "format": "epoch_millis"    <3>
+    }
+  ]
+}
+----
+
+<1> Wildcard patten used to match field names, specified as a string.
+<2> Wildcard patten used to match field names, specified as an object.
+<3> With the object notation, you can use the `format` parameter to specify a
+    format for the field's returned doc values. <<date,Date fields>> support a
+    <<mapping-date-format,date `format`>>. <<number,Numeric fields>> support a
+    https://docs.oracle.com/javase/8/docs/api/java/text/DecimalFormat.html[DecimalFormat
+    pattern]. Other field datatypes do not support the `format` parameter.
+=====
+
+The `docvalue_fields` parameter has some limitations:
+
+* Doc values are stored for supported fields by default. However, <<text,`text`
+fields>> do not support doc values.
+
+*  If you use the `docvalue_fields` parameter to specify a field without a doc
+value, the search attempts to load the value from the <<modules-fielddata,field
+data cache>>. This causes the field's terms to be loaded into memory, resulting
+in higher memory consumption.
+
+*  You cannot use the `docvalue_fields` parameter to retrieve doc values for
```

Comment:
Small comment, this could be better as a 'tip', because it's not really a limitation (we have a great way to do this, and it's through `inner_hits`).